src/textual/_compositor.py

```diff
@@ -0,0 +1,549 @@
+"""
+
+The compositor handles combining widgets in to a single screen (i.e. compositing).
+
+It also stores the results of that process, so that Textual knows the widgets on
+the screen and their locations. The compositor uses this information to answer
+queries regarding the widget under an office, or the style under an offset.
+
+Additionally, the compositor can render portions of the screen which may have updated,
+without having to render the entire screen.
+
+"""
+
+from __future__ import annotations
+
+from operator import attrgetter, itemgetter
+import sys
+from typing import cast, Callable, Iterator, Iterable, NamedTuple, TYPE_CHECKING
+
+import rich.repr
+from rich.console import Console, ConsoleOptions, RenderResult
+from rich.control import Control
+from rich.segment import Segment, SegmentLines
+from rich.style import Style
+
+from . import errors, log
+from .geometry import Region, Offset, Size
+
+
+from ._loop import loop_last
+from ._types import Lines
+from .widget import Widget
+
+if sys.version_info >= (3, 10):
+    from typing import TypeAlias
+else:  # pragma: no cover
+    from typing_extensions import TypeAlias
+
+
+if TYPE_CHECKING:
+    from .screen import Screen
+    from .widget import Widget
+
+
+class ReflowResult(NamedTuple):
+    """The result of a reflow operation. Describes the chances to widgets."""
+
+    hidden: set[Widget]  # Widgets that are hidden
+    shown: set[Widget]  # Widgets that are shown
+    resized: set[Widget]  # Widgets that have been resized
+
+
+class RenderRegion(NamedTuple):
+    """Defines the absolute location of a Widget."""
+
+    region: Region  # The region occupied by the widget
+    order: tuple[int, ...]  # A tuple of ints defining the painting order
+    clip: Region  # A region to clip the widget by (if a Widget is within a container)
+    virtual_size: Size  # The virtual size  (scrollable region) of a widget if it is a container
+    container_size: Size  # The container size (area no occupied by scrollbars)
+
+
+RenderRegionMap: TypeAlias = "dict[Widget, RenderRegion]"
+
+
+@rich.repr.auto
+class LayoutUpdate:
+    """A renderable containing the result of a render for a given region."""
+
+    def __init__(self, lines: Lines, region: Region) -> None:
+        self.lines = lines
+        self.region = region
+
+    def __rich_console__(
+        self, console: Console, options: ConsoleOptions
+    ) -> RenderResult:
+        yield Control.home()
```

Comment:
Do we need this? (`move_to` moves to an absolute position anyway)

Reply:
Possibly not!

Reply:
You're right. Removed.